src/main/java/databaseInteracting/DataBaseInteracting.java

```diff
@@ -0,0 +1,173 @@
+   *                          prepared statement of the query to list all products.
+   * @return resultSet with data of the products table.
+   * @throws SQLException - Exception of database.
+   */
+  public ResultSet createResultSet(PreparedStatement preparedStatement) throws SQLException {
+    return preparedStatement.executeQuery();
+  }
+
+  /**
+   * Method to return all products into a Products table.
+   *
+   * @return list of products.
+   * @throws SQLException Exception of database.
+   */
+  public List<databaseInteracting.Product> searchForAllProductsOnDatabase() throws SQLException {
+    List<databaseInteracting.Product> list = new ArrayList<>();
+    try (PreparedStatement preparedStatement = createPrepareStatementToProductList();
+         ResultSet aResultSet = createResultSet(preparedStatement)) {
+      PrepareAndExecuteMethod(aResultSet, list);
+    }
+    return list;
+  }
+  /*
+  /**
+   * Method to make a query into the shopping cart table.
+   * @return a list of products from shopping cart table.
+   * @throws SQLException - Exception of database.
+  */
+  public ResultSet searchForAllProductsOnShoppingCart(int cartId) throws SQLException {
+    String sql = "SELECT p.id, p.name, p.price, l.quantity FROM products p," +
+        " shoppinglist l, shoppingcart c where p.id = l.idproduct AND l.idcart = ?;";
+      PreparedStatement preparedStatement = connection.prepareStatement(sql);
+      preparedStatement.setInt(1,cartId);
+      return preparedStatement.executeQuery();
+  }
+
+  /**
+   * Method to calculate the total amount of the sale.
+   *
+   * @param shoppingCartId int - id of the shopping cart.
+   * @return result set with a float value of total amount of sale.
+   * @throws SQLException - Exception related of database.
+   */
+  public ResultSet calculateTotalAmount(int shoppingCartId) throws SQLException {
+    PreparedStatement preparedStatement = connection.prepareStatement(
+        "SELECT SUM (p.price * l.quantity) AS totalSum FROM  products p," +
+            " shoppinglist l, shoppingcart c WHERE l.idProduct = p.id AND c.idShoppingCart = ?;");
+    preparedStatement.setInt(1, shoppingCartId);
+    return preparedStatement.executeQuery();
+  }
+
+  /**
+   * Method to insert a product into a shopping cart.
+   *
+   * @param idProduct int - id of product selected by the client.
+   * @param quantity  int - quantity of the product selected by the client.
+   * @throws SQLException - Exception to database.
+   */
+  public void insertProductIntoAShoppingCartTable(int idCart, int idProduct, int quantity) throws SQLException {
+    String sql = "INSERT INTO shoppinglist (idcart, idproduct, quantity) values (?, ?, ?)";
+    PreparedStatement preparedStatement = connection.prepareStatement(sql);
+    preparedStatement.setInt(1, idCart);
+    preparedStatement.setInt(2, idProduct);
+    preparedStatement.setInt(3, quantity);
+    preparedStatement.executeUpdate();
+    preparedStatement.close();
```

Comment:
Isnt better to use try with resources?

Reply:
Tried to use a try with resource, but not work. I'll try to do this change later.